manim/utils/ipython_magic.py

```diff
@@ -128,21 +130,10 @@ def construct(self):
                 config.digest_args(args)
 
                 renderer = None
-                if config.renderer == "opengl":
-                    # Check if the imported mobjects extend the OpenGLMobject class
-                    # meaning ConvertToOpenGL did its job
-                    if "OpenGLMobject" in [
-                        parent_class.__name__ for parent_class in Group.mro()
-                    ]:
-                        from manim.renderer.opengl_renderer import OpenGLRenderer
-
-                        renderer = OpenGLRenderer()
-                    else:
-                        logger.warning(
-                            "Renderer must be set to OpenGL in the configuration file "
-                            "before importing Manim! Using cairo renderer instead.",
-                        )
-                        config.renderer = "cairo"
+                if config.renderer == RendererType.OPENGL:
+                    from manim.renderer.opengl_renderer import OpenGLRenderer
+
+                    renderer = OpenGLRenderer()
```

Comment:
The check was added in a very early iteration of `ConvertToOpenGL`, when changing `config.renderer` after importing the library was somewhat less intelligent (or brutal, whatever you want to call it) so that it didn't change the MROs of already imported class objects -- using OpenGL in Jupyter notebooks was only possible by setting the renderer to OpenGL in the config file.

Since then, the implementation has improved and this restriction is no longer required -- and so is the check. If `config.renderer` is set to OpenGL, the mobject inheritance is guaranteed to have been changed to the OpenGL version.